server/src/main/java/org/elasticsearch/reservedstate/service/ReservedStateUpdateTask.java

```diff
@@ -134,7 +134,14 @@ private void checkAndThrowOnError(List<String> errors, ClusterState currentState
                 ReservedStateErrorMetadata.ErrorKind.VALIDATION
             );
 
-            errorReporter.accept(currentState, errorState);
+            /*
+             * It doesn't matter this reporter needs to re-access the base state,
+             * any updates set by this task will just be discarded when the below exception is thrown,
+             * and we just need to set the error state once
+             *
+             * TODO - does this need to use a 'force' option on the underlying ReservedErrorStateTask?
```

Comment:
Can you explain what you mean by force, do you mean update the state regardless of the version check?

Reply:
This method previously had a `currentState` variable that it compared against, containing some modifications that were applied by `ReservedStateUpdateTaskExecutor`. The error state will now be compared against the actual live cluster state. Is there a chance this could cause errors that should be applied to be ignored?

eg, if the modified state went from error1 -> s2 -> error2, but the error test is now going from error1 -> error2, do we need to force error2 to be applied on top of error1 to maintain the current behaviour, or does this not actually matter here?

Reply:
Oh that's totally fine, we don't care about changes to the cluster state applied by the update task executor if they failed. It's OK to take the latest state without worrying what `ReservedStateUpdateTaskExecutor` produced. The thing is though, if what we get back from the `ReservedStateUpdateTaskExecutor` as error version is equal or newer than ours, we can bail immediately, no need to start another cluster state update task.